### PR TITLE
Add Red Team PII Tests

### PR DESCRIPTION
**Pull request by mldangelo**
## Summary
This PR adds tests for Personally Identifiable Information (PII) leaks and updates relevant documentation. It includes the following changes:

### Changes
1. **New PII Test Category:**
   - Implemented a new module `getPiiTests.ts` to generate adversarial tests that ensure the model does not inadvertently disclose PII.
   - Defined multiple PII test categories, including `DirectPIIRequests`, `SessionBasedLeaks`, `SocialEngineeringPrompts`, and `APIAndDatabaseAccessRequests`. 
   - Provided detailed examples and descriptions for each category to aid in understanding and implementation.
   
2. **Documentation Updates:**
   - Updated the title capitalization for consistency in `llm-redteaming.md`.
   - Added a new section on PII leaks to the list of test capabilities.
   - Improved formatting for code snippets by specifying `sh` for shell commands.
   - Updated the order and descriptions of adversarial test categories for clarity.

3. Runs `npx prettier --write` over the modified files. 

### Test Instructions

1. Create a new red team project and generate adversarial test cases using the following commands:
   ```sh
   npx promptfoo init my-redteam-project
   cd my-redteam-project
   npx promptfoo generate redteam -w --plugins pii
   npx promptfoo eval
   ```

3. Verify that PII leak tests are included in the generated test cases and that they function as expected.

### CODE REVIEW

File: src/redteam/getPiiTests.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+  const prompts = generatedPrompts
+    .split('\n')
+    .filter((line) => line.includes('Prompt:'))
+    .map((line) => line.substring(line.indexOf('Prompt:') + 'Prompt:'.length).trim());
+
+  return prompts.map((prompt) => ({
+    vars: {
+      [injectVar]: prompt,
+      name,
```
Comment (typpo) [inline src/redteam/getPiiTests.ts]: We don't have to include `name` as a var, since it is placed directly in the prompt text by nunjucks

**Review by typpo — APPROVED** [review]
Looks great!

File: src/redteam/getPiiTests.ts (hunk 1 above)
Comment (mldangelo) [inline src/redteam/getPiiTests.ts]:
```suggestion
```

**Comment by typpo** [comment]
Very nice!

**Pull request merged**